### PR TITLE
Show skip link on each challenge

### PR DESCRIPTION
**Pull request by wakqasahmed**
Show skip to next challenge on each waypoint and bonfires.

screen shot 2016-01-03 at 1 01 50 am

### CODE REVIEW

**Comment by SaintPeter** [comment]
While this could work, it's a bit rough for production.
I'd prefer to see a small control block under the "reset/help/bug" section with both "next" and "prev" and appropriate icons.  It can't be too tall, though, maybe half height of the above buttons?

**Comment by wakqasahmed** [comment]
screen shot 2016-01-03 at 11 59 07 pm

File: .gitignore [hunk 1]
```diff
@@ -40,7 +40,6 @@ server/manifests/*
 !server/manifests/README.md
 
 public/js/main*
-public/js/commonFramework*
 public/js/sandbox*
```
Comment (SaintPeter) [inline .gitignore]: What is this all about?  the `/public/js/*` files are GENERATED files, the source files are broken out into a bunch of different files under `/client/commonFramework`.

You need to make you changes to the appropriate sub file.

**Comment by SaintPeter** [comment]
Ok, looks like some serious issues here.  The changes you made to `commonFramework.js` need to be made in the constituent parts of `public/commonFramework/`.

I DO like the general look and feel, although maybe a contrasting button color would be better?

File: server/views/coursewares/showJS.jade [hunk 2]
```diff
@@ -3,7 +3,7 @@ block content
     link(rel='stylesheet', href='/bower_components/CodeMirror/lib/codemirror.css')
     link(rel='stylesheet', href='/bower_components/CodeMirror/addon/lint/lint.css')
     link(rel='stylesheet', href='/bower_components/CodeMirror/theme/monokai.css')
-    link(rel='stylesheet', href='/css/ubuntu.css')
+    link(rel='stylesheet', href='/css/ubuntu.css')
```
Comment (ltegman) [inline server/views/coursewares/showJS.jade]: remove trailing spaces

File: server/views/coursewares/showBonfire.jade [hunk 3]
```diff
@@ -4,7 +4,6 @@ block content
     link(rel='stylesheet', href='/bower_components/CodeMirror/addon/lint/lint.css')
     link(rel='stylesheet', href='/bower_components/CodeMirror/theme/monokai.css')
     link(rel='stylesheet', href='/css/ubuntu.css')
-
     .row
```
Comment (ltegman) [inline server/views/coursewares/showBonfire.jade]: break between scripts and other markup should be kept.

**Comment by wakqasahmed** [comment]
btn-info or btn-primary seems contrasting, which one do you suggest?

**Comment by wakqasahmed** [comment]
Also, I found font problem in all those buttons, fixing them too.

screen shot 2016-01-04 at 1 14 36 am

**Comment by SaintPeter** [comment]
The file `public/js/commonFramework.js` should not be part of the commit.  I think it was leftover from when you had removed it from the `.gitignore.`

I wonder what `.btn-warning` might look like.  We use it in the step challenges.

**Comment by BerkeleyTrue** [comment]
This PR will need @QuincyLarson's approval of the UI before it can be accepted.

**Comment by QuincyLarson** [comment]
This is an interesting feature. Have very many people asked for this functionality?
My concern is that it adds even more buttons to an already cluttered user interface. And paging through our challenges by clicking the next and previous buttons may be quite slow in production. 
I'd like to wait until we've reactified our challenges before we consider incorporating this type of feature. When we do, we will use your implementation as the model.

**Comment by SaintPeter** [comment]
@QuincyLarson This is a pretty commonly asked for feature.  I personally would use it extensively, although my use case is maybe not identical to others.  I know the performance would not be ideal, currently, but I don't see the harm in implementing now.

I don't see it add too much to "clutter", since it will be visually distinct and part of the "control block".

**Comment by wakqasahmed** [comment]
I agree with @SaintPeter 

@QuincyLarson Regarding production page performance I think i can combine the functions PrevChallenge and NextChallenge to process once and provide both links, this will keep the performance as it is now but serve our purpose beautifully.

Let me know if you are interested.

**Comment by BerkeleyTrue** [comment]
After discussing with @QuincyLarson We have decided to wait until the react version of the site to implement this feature. The react version of the challenges is coming soon. Thanks for you're patience.

Happy Coding!

**Pull request closed without merging**